.github/workflows/update-api-files.yml

```diff
@@ -0,0 +1,78 @@
+name: Update API files
+
+on:
+  workflow_dispatch:
```

Comment:
This is a nice one!
How do people trigger this when they want to generate new api.md? I guess they will need to run this manually and choose the right branch for themselves?

Reply:
Correct, they have to run manually through the github actions UI and select their branch all manually. This is mostly just so I can make use of it haha, I still think all our documentation should say to simply build locally to build the API files -- but if we want to make PRs easier perhaps we should add a PR label

Reply:
I'd say add a PR label. Kicking via actions tab is not all that different from adding a PR label. But latter is more discoverable.

We'd still review the API changes in the same way.

Reply:
Agree with @prprabhu-ms, that would be much faster for contributor to run the pipeline, but we need another label for this like 'api-change'

Reply:
I will complete this PR and we can make that a seperate PR (as we would also want things like a comment when the build fails due to api changes to highlight the label is available)